src/test/ut/ut_global_kvs.c

```diff
@@ -0,0 +1,200 @@
+
+struct test_key {
+        char type;
+        str256_t name;
+} test_key;
+
+/* Set the global kvs */
+static void test_nsal_set_global_kvs(void)
+{
+	struct collection_item *errors = NULL;
+	struct collection_item *cfg_items = NULL;
+	int rc = 0;
+
+	rc = log_init("/var/log/cortx/fs/cortxfs.log", LEVEL_DEBUG);
+	if (rc != 0) {
+		rc = -EINVAL;
+		printf("Log init failed, rc: %d\n", rc);
+		goto out;
+	}
+
+	rc = config_from_file("libcortxfs", DEFAULT_CONFIG, &cfg_items,
+			      INI_STOP_ON_ERROR, &errors);
+	if (rc) {
+		printf("Can't load config rc = %d", rc);
+		rc = -rc;
+		goto out;
+	}
+
+	rc = nsal_module_init(cfg_items);
+	if (rc) {
+		printf("nsal init failed rc = %d", rc);
+		rc = -rc;
+		goto out;
+	}
+
+	/* make use of kvs_fid as global kvs */
+	rc = nsal_set_global_kvs("<0x780000000000000b:1>");
+	if (rc) {
+		printf("nsal_set_global_kvs failed rc = %d", rc);
+		rc = -rc;
+		goto out;
+	}
+
+out:
+	if (rc) {
+		free_ini_config_errors(errors);
+	}
+
+	ut_assert_int_equal(rc, 0);
+}
+
+
+/**
+ * Test to add a KV pair in the global KVStore.
+ */
+static void test_nsal_global_set_kv(void)
+{
+	int rc = 0;
+	char *name = "component";
+	char *value = "cortxfs";
+	struct test_key *key1;
+
+	rc = nsal_global_alloc((void **)&key1, sizeof(struct test_key));
+	ut_assert_int_equal(rc, 0);
+
+	key1->type = 'T';
+	str256_from_cstr(key1->name, name, strlen(name));
+
+	rc = nsal_global_set_kv(key1, sizeof(struct test_key),
+				 (void*)value, strlen(value)+1);
+	ut_assert_int_equal(rc, 0);
+
+	nsal_global_free(key1);
+}
+
+/**
+ * Test to add a KV pair in the global KVStore.
+ */
+static void test_nsal_global_get_kv(void)
+{
+	int rc = 0;
+	char *name = "component";
+	char *value = "cortxfs";
+	char *val = NULL;
+	struct test_key *key1;
+	size_t vlen=0;
+
+	rc = nsal_global_alloc((void **)&key1, sizeof(struct test_key));
+	ut_assert_int_equal(rc, 0);
+
+	key1->type = 'T';
+	str256_from_cstr(key1->name, name, strlen(name));
+
+	rc = nsal_global_get_kv(key1, sizeof(struct test_key), (void**)&val,
+			    &vlen);
+	ut_assert_int_equal(rc, 0);
+
+	rc = strcmp(value, val);
+	ut_assert_int_equal(rc, 0);
+
+	nsal_global_free(key1);
+}
+
+/**
+ * Test to add a KV pair in the global KVStore.
+ */
+static void test_nsal_global_del_kv(void)
+{
+	int rc = 0;
+	char *name = "component";
+	char *val = NULL;
+	size_t vlen=0;
+	struct test_key *key1;
+
+	rc = nsal_global_alloc((void **)&key1, sizeof(struct test_key));
+	ut_assert_int_equal(rc, 0);
+
+	key1->type = 'T';
+	str256_from_cstr(key1->name, name, strlen(name));
+
+	rc = nsal_global_del_kv(key1, sizeof(struct test_key));
+	ut_assert_int_equal(rc, 0);
+
+	/* try to retrive the deleted key */
+	rc = nsal_global_get_kv(key1, sizeof(struct test_key), (void**)&val,
+			    &vlen);
+	ut_assert_int_equal(rc, -ENOENT);
+
+	nsal_global_free(key1);
+}
+
+
+int main(void)
+{
+        int rc = 0;
+        char *test_logs = "/var/log/cortx/test/ut/ut_nsal.logs";
+
+	printf("Global KVS Tests\n");
```

Comment:
Please check the indentation in lines 164 165 and 167

Reply:
rectified it.